src/libs/memoize/cache/ArrayCache.ts

```diff
@@ -0,0 +1,82 @@
+import type {Cache, CacheConfig} from './types';
+
+/**
+ * Builder of the cache using `Array` primitive under the hood. It is an LRU cache, where the most recently accessed elements are at the end of the array, and the least recently accessed elements are at the front.
+ * @param config - Cache configuration, check `CacheConfig` type for more details.
+ * @returns
+ */
+function ArrayCache<K, V>(config: CacheConfig<K>): Cache<K, V> {
+    const cache: Array<[K, V]> = [];
+
+    const {maxSize, keyComparator} = config;
+
+    function getKeyIndex(key: K) {
+        // We search the array backwards because the most recently added entries are at the end, and our heuristic follows the principles of an LRU cache - that the most recently added entries are most likely to be used again.
+        for (let i = cache.length - 1; i >= 0; i--) {
+            if (keyComparator(cache[i][0], key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    return {
+        get(key) {
```

Comment:
```suggestion
        get(key: K) {
```

Reply:
Types are provided by `Cache` interface, which `ArrayCache` implements so it would be redundant https://github.com/Expensify/App/blob/768bf3ae95b9b3ac5b88bfdde759c0b9313d2bab/src/libs/memoize/cache/types.ts#L8